app/icon.js

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
```

Comment:
> the switch case would conflate all the possible icon combinations.

I agree on that, I found that particular solution a bit weird when I first read about it a while ago, but thought that "this is how they do it in React world" :shrug: ;)

I'll fix it.